ledger/metrics/src/test/scala/com/daml/telemetry/TelemetryContextSpec.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.telemetry
+
+import io.opentelemetry.api.trace.Span
+import org.scalatest.Assertion
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+/** Other cases are covered by [[TelemetrySpec]] */
+class TelemetryContextSpec extends AnyWordSpecLike with Matchers {
+
+  "DefaultTelemetryContext.runInOpenTelemetryScope" should {
```

Comment:
I tried to increase the coverage here but failed. If you have an idea on how to test it further, please let me know. I'm also not sure if this spec is not pointless now...

Reply:
This unit test only checks that the body is run but there are 2 more things to check for `runInOpenTelemetryScope` implementations: that they set the scope in the current context and that the scope is closed after the body has run, even if an exception is thrown.

Reply:
Also, `encodeMetadata` is currently not tested at all.

Reply:
Yup, I know. The problem is how to capture these. The API doesn't expose too much.

Reply:
You could slightly change the API to optionally allow injecting a `Context` and then mock/fake it to detect setting a scope. You should also be able to easily mock a `Scope`.

Reply:
I wanted to avoid that, as the problem is that we don't have a single method that exposes the `opentelemetry` API, so far.

Reply:
How about overloading just for the tests and restricting access control?

Reply:
It should be enough to make the `Tracer` instance injectable into the `TelemetryContext` for testing. I.e., then you can use an in-memory exporter to check what spans have been created for the various test cases (just as you did for `Telemetry`). I would suggest focusing on testing `TelemetryContext` with OpenTelemetry components and not mocking any of the tracing related components, as our primary goal is to prove that OpenTelemetry picks up the created spans via the `TelemetryContext` abstraction.

Reply:
I find it a bad practice to add stuff just for test purposes to the production code, obviously. But if you are fine with me adding it, sure.

Reply:
By allowing injecting the `Tracer` you are just removing dependency on a global object from `TelemetryContext` which is good design anyway.

Reply:
Yup, initially I even did this. Then, I got inspired by some JUnit 5 extension ;) Let me make that change, thanks!

Reply:
@fabiotudone-da @miklos-da, done. I didn't have access to the `Scope` but the coverage is much better now.